app/assets/javascripts/app/views/tag_following_list_view.js

```diff
@@ -63,12 +63,36 @@ app.views.TagFollowingList = app.views.Base.extend({
   createTagFollowing: function(evt) {
     if(evt){ evt.preventDefault(); }
 
-    this.collection.create({"name":this.$(".tag_input").val()});
+    var name = this.$(".tag_input").val();
+    // compare tag_text_regexp in app/models/acts_as_taggable_on-tag.rb
+    var normalizedName = (name === "<3" ? name : name.replace(
+        new RegExp("[^" + PosixBracketExpressions.alnum + "_\\-]+", "gi"), "").toLowerCase());
+
+    this.collection.create({"name":normalizedName});
+
     this.$(".tag_input").val("");
     return this;
   },
 
   appendTagFollowing: function(tag) {
+    // we want the tags sorted. Not ideal on pageload but we need this on create
```

Comment:
You already mentioned that this is bad on pageload. It would be great if you could add a new function that adds all tags on pageload and use it in `postRenderTemplate`.